test/e2e/validation_test.go

```diff
@@ -667,14 +665,12 @@ func (tc *testContext) findNodeCSRs(nodeName string) ([]certificates.Certificate
 
 // validateUpgradeableCondition ensures that the operator's Upgradeable condition is correctly communicated to OLM
 func (tc *testContext) validateUpgradeableCondition(expected meta.ConditionStatus) error {
-	ocName, present := os.LookupEnv(condition.OperatorConditionName)
-	if !present {
-		// Implies operator is not OLM-managed
-		return nil
-	}
-	err := wait.Poll(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
-		oc := &operators.OperatorCondition{}
-		err := tc.client.Cache.Get(context.TODO(), types.NamespacedName{Namespace: tc.namespace, Name: ocName}, oc)
+	ocName, err := tc.getOperatorConditionName()
+	if err != nil || ocName == "" {
+		return errors.Errorf("unable to get operatorCondition name from namespace %s", tc.namespace)
+	}
+	err = wait.Poll(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
```

Comment:
Is it required to wait first? Can we use `wait.PollImmediate` instead to avoid the idle time?

Reply:
yes it is required to wait, we are not exactly sure at what point OLM injects the variable. I tried pollImmediate and it did not work in this case.